yammer-metrics/src/main/java/com/wavefront/integrations/metrics/SocketMetricsProcessor.java

```diff
@@ -122,7 +113,15 @@ public void processMeter(MetricName name, Metered meter, Void context) throws Ex
   @Override
   public void processCounter(MetricName name, Counter counter, Void context) throws Exception {
     if (!sendZeroCounters && counter.count() == 0) return;
-    writeMetric(name, null, counter.count());
+
+    // handle delta counters
+    if (counter instanceof DeltaCounter) {
+      long count = counter.count();
+      writeMetric(name, null, count);
+      counter.dec(count);
```

Comment:
There are bunch of places where we 
1) get the count, 
2) do something and 
3) decrement the delta counter

if we flip 2) and 3)
you can add a static util method

public static long processDeltaCounter(Counter counter) {
  long count = counter.count();
  counter.dec(count);
  return count;
}
and that way user does not have to worry about this 2 step approach and prevent inadvertent bugs.

Reply:
Added a static method.